sql/hive/src/test/scala/org/apache/spark/sql/sources/HadoopFsRelationTest.scala

```diff
@@ -515,20 +515,20 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils with Tes
         .save(subdir.getCanonicalPath)
 
       // Inferring schema should throw error as it should not find any file to infer
-      val e = intercept[Exception] {
-        spark.read.format(dataSourceName).load(dir.getCanonicalPath)
-      }
-
-      e match {
-        case _: AnalysisException =>
-          assert(e.getMessage.contains("infer"))
+      if (dataSourceName != classOf[SimpleTextSource].getCanonicalName) {
```

Comment:
Simply `dataSourceName != classOf[SimpleTextSource].getCanonicalName` checking is added here because text data source dose not throw an exception anymore if `dataSchema` is not given.